students/chelsea_nayan/lesson05/database.py

```diff
@@ -0,0 +1,166 @@
+'''Adds products in a csv file to a MongoDB database'''
+# lesson 05: Consuming APIs
+
+# Checklist:
+# Create a product db with attributes that reflect the contents of the csv file
+# Import all data in the csv files into your MongoDB implementation
+# Write queries to retrieve the product database
+# Write a query to integrate customer and product data
+# no errors from pylint
+# quantity_available of '0' is understood as 'not available'
+# Create a test_database.py
+
+# pylint: disable=too-many-statements
+# pylint: disable=invalid-name
+# pylint: disable=too-many-locals
+
+import logging
+import csv
+import os
+from pymongo import MongoClient
+# Set up logger
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+class MongoDBConnection():
+    '''MongoDBConnection'''
+
+    def __init__(self, host='127.0.0.1', port=27101):
+        '''Be sure to use the ip address, not the name for local windows, intializes'''
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        '''Enters connection?'''
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        '''Exits connection'''
+        self.connection.close()
+
+def import_data(directory_name, product_file, customer_file, rentals_file):
+    '''
+    Populate new MongoDB database using the three csv file inputs.
+    Returns two tuples (1) record count of # of products, customers, rentals
+                       (2) count of any errors that occured
+    '''
+    product_error, customer_error, rentals_error = 0, 0, 0
+    product_file_path = os.path.join(directory_name, product_file)
+    customer_file_path = os.path.join(directory_name, customer_file)
+    rentals_file_path = os.path.join(directory_name, rentals_file)
+
+    mongo = MongoDBConnection()
+    with mongo:
+        db = mongo.connection.media
+        products = db['products']
+        customers = db['customers']
+        rentals = db['rentals']
+
+    # Attempt to import product data file into MongoDB db, create product collection
+    try:
+        with open(product_file_path, encoding='utf-8-sig') as file:
+            reader = csv.DictReader(file)
+            for row in reader:
+                add_product = {'_id': row['_id'],
+                               'description': row['description'],
+                               'product_type': row['product_type'],
+                               'quantity_available': row['quantity_available']}
+                try:
+                    products.insert_single(add_product)
```

Comment:
There is no _insert_single_ method for a collection object. Check for the method used to add a single record (name is pretty similar).